### PR TITLE
Configure Dependabot

### PR DESCRIPTION
**Pull request by nathanhammond**
This PR configures Dependabot to monitor dependencies across all of our own packages. It does not attempt to set up Dependabot for our `examples` or our `create-turbo` templates. (I want to address that update separately.)

Note: Dependabot does not officially support `pnpm` (dependabot/dependabot-core#1736) and is not workspaces-friendly (dependabot/dependabot-core#4993). All Dependabot PRs for the `npm` ecosystem *will* require checking out, regenerating the lockfile, and pushing back to the branch.

I've configured checks to be monthly which seems like a decent cadence to monitor things (and cluster everything to a few hours required to shepherd everything through).

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -5,7 +5,31 @@ updates:
     directory: '/'
     schedule:
       interval: 'monthly'
-    ignore:
```
Comment (gsoltis) [inline .github/dependabot.yml:8]: Is this safe to remove? The comment seems to indicate that it's there for some reason.

**Review by gsoltis — DISMISSED** [review]
This all seems fine, I just don't know what the implications of removing the `ignore` block are.

File: .github/dependabot.yml (hunk 1 above)
Comment (nathanhammond) [reply .github/dependabot.yml:8]: We're not using Husky anymore. It's nowhere to be found in our lockfile.

**Comment by samnap11** [comment]
I'm sorry if this might be bringing a new question/problem to the table. However, I would like to suggest an alternative to Dependabot for monitoring dependencies, which is Renovate. It supports [pnpm](https://docs.renovatebot.com/javascript/), [Go Modules](https://docs.renovatebot.com/golang/), and also [pnpm-workspace](https://github.com/renovatebot/renovate/pull/10071).

Hopefully, with Renovate, the process of bumping and monitoring dependencies can be performed fully by the bot.

**Comment by nathanhammond** [comment]
@samnap11 Thanks for the pointer! Integrating Renovate is definitely something we could consider in the future. For now this is an intentionally targeted change which addresses our existing integration with Dependabot.

**Review by gsoltis — APPROVED** [review]

**Comment by JoA-MoS** [comment]
As an FYI: I had a similar experience but noticed some of the PR's were correct. The PR's that were correct didn't have the 
`in /<sub-dir>` text.

ie:
```
chore(deps): bump commander from 9.2.0 to 9.3.0 in /my/sub-project
``` 
but PR's with just text like
```
chore(deps): bump commander from 9.2.0 to 9.3.0
```
updated all instances of commander in the project.json and the pacakge-lock.json at the root of the project

so maybe define the directory as `/` and don't define the individual subdirectories.

**Pull request merged**